server/routes/get_agent.ts

```diff
@@ -14,10 +14,10 @@ export const getAgent = async (id: string, client: OpenSearchClient['transport']
       path,
     });
 
-    if (!response || !response.body.configuration?.agent_id) {
+    if (!response || !response.body.ml_configuration?.agent_id) {
       throw new Error(`cannot get agent ${id} by calling the api: ${path}`);
     }
-    return response.body.configuration.agent_id;
+    return response.body.ml_configuration.agent_id;
```

Comment:
maybe add a fallback here to make it compatible with existing data
```suggestion
    return response.body.ml_configuration.agent_id || response.body. configuration.agent_id;
```

Reply:
The code will throw error if `!response.body.ml_configuration?.agent_id` above. I think `response.body. configuration.agent_id` won't be executed. Need to find some way to compatible with legacy data.

Reply:
yes, the if condition should change accordingly.
```typescript
if (!response || !response.body.ml_configuration?.agent_id || !response.body.configuration?.agent_id) {
      throw new Error(`cannot get agent ${id} by calling the api: ${path}`);
    }
```

Reply:
I changed the condition of throwing error as well.

Reply:
@Hailong-am  Maybe && between the configuration since either agent_id can be retrieved will be ok